framework/sandstone.cpp

```diff
@@ -1021,13 +1046,23 @@ static void commit_shmem()
         close(sApp->shmemfd);
         sApp->shmemfd = -1;
     }
+    sApp->shmem->total_cpu_count = num_cpus();
 }
 
 static void attach_shmem(int fd)
 {
     assert(sApp->current_fork_mode() == SandstoneApplication::child_exec_each_test);
 
-    int size = ROUND_UP_TO_PAGE(sizeof(SandstoneApplication::SharedMemory));
+    size_t size;
+    if (struct stat st; fstat(fd, &st) >= 0) {
```

Comment:
hm, I don't see this as a good coding style.

Reply:
Since `st` isn't used beyond the `if`, I prefer it.